### PR TITLE
feat(overlay): add connected overlay directive

### PR DESCRIPTION
**Pull request by jelbourn**
R: @kara @hansl 

This is the initial version of the final part blocking things like menus and select. It's a directive that creates an overlay from the content in a template and positions it connected to some other element.

### CODE REVIEW

File: src/core/overlay/overlay-directives.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            {originX: this.positions[0].overlayX, originY: this.positions[0].originY},
+            {overlayX: this.positions[0].overlayX, overlayY: this.positions[0].overlayY});
+
+    this._overlay.create(overlayConfig).then(ref => {
+      this._overlayRef = ref;
+      this._overlayRef.attach(this._templatePortal);
+    });
+  }
+
+  /** Destroys the overlat created by this directive. */
```
Comment (justindujardin) [inline src/core/overlay/overlay-directives.ts]: typo: overlat

File: src/core/overlay/overlay-directives.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      {originX: 'start', originY: 'bottom'},
+      {overlayX: 'start', overlayY: 'top'}),
+  new ConnectionPositionPair(
+      {originX: 'start', originY: 'top'},
+      {overlayX: 'start', overlayY: 'bottom'}),
+];
+
+
+/**
+ * Directive to facilitate declarative creation an Overlay using a ConnectedPositionStrategy.
```
Comment (justindujardin) [inline src/core/overlay/overlay-directives.ts]: missing 'of'

File: src/demo-app/overlay/overlay-demo.ts [hunk 3]
```diff
@@ -1,36 +1,32 @@
...
-
-
-@Directive({selector: '[overlay-origin]'})
-class OverlayOrigin {
-  constructor(public elementRef: ElementRef) { }
-}
+import {
+    ConnectedOverlayDirective,
+    OverlayOrigin
+} from '@angular2-material/core/overlay/overlay-directives';
```
Comment (kara) [inline src/demo-app/overlay/overlay-demo.ts]: Should we be exporting these in `core.ts` so we don't have to go down the path?

File: src/demo-app/overlay/overlay-demo.ts [hunk 4]
```diff
@@ -1,36 +1,32 @@
...
 
 
 @Component({
   moduleId: module.id,
   selector: 'overlay-demo',
   templateUrl: 'overlay-demo.html',
   styleUrls: ['overlay-demo.css'],
-  directives: [TemplatePortalDirective, OverlayOrigin],
+  directives: [TemplatePortalDirective, OverlayOrigin, ConnectedOverlayDirective],
```
Comment (kara) [inline src/demo-app/overlay/overlay-demo.ts]: What do you think about putting `OverlayOrigin` and `ConnectedOverlayDirective` in one of our shouty arrays?  `MD_OVERLAY_DIRECTIVES`?

**Comment by kara** [comment]
Added a few questions, but LGTM

File: src/demo-app/overlay/overlay-demo.ts (hunk 3 above)
Comment (jelbourn) [inline src/demo-app/overlay/overlay-demo.ts]: Done

File: src/demo-app/overlay/overlay-demo.ts (hunk 4 above)
Comment (jelbourn) [inline src/demo-app/overlay/overlay-demo.ts]: Now exporting `OVERLAY_DIRECTIVES`

**Comment by jelbourn** [comment]
@justindujardin @kara addressed comments

**Pull request merged**